modules/custom/social_graphql/src/Plugin/GraphQL/Schema/OpenSocialBaseSchema.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Drupal\social_graphql\Plugin\GraphQL\Schema;
+
+use Drupal\graphql\GraphQL\ResolverBuilder;
+use Drupal\graphql\GraphQL\ResolverRegistryInterface;
+use Drupal\graphql\Plugin\GraphQL\Schema\SdlSchemaPluginBase;
+use Drupal\social_graphql\GraphQL\ResolverRegistry;
+
+/**
+ * The provider of the schema base for the Open Social GraphQL API.
+ *
+ * Doesn't do anything itself but provides a target for GraphQL Schema
+ * extensions. Schema Extensions should implement `SdlSchemaExtensionPluginBase`
+ * and should not subclass this class.
+ *
+ * This class borrows from the ComposableSchema example but intentionally does
+ * not implement the extension configuration that that schema provides. Instead
+ * the SdlSchemaPluginBase loads the schema extensions for all Open Social
+ * features that are enabled.
+ *
+ * @Schema(
+ *   id = "open_social",
+ *   name = "Open Social GraphQL Schema"
+ * )
+ */
+class OpenSocialBaseSchema extends SdlSchemaPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getResolverRegistry() {
+    return new ResolverRegistry();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getSchema(ResolverRegistryInterface $registry) {
+    // Add Open Social base types to the schema.
+    $this->getBaseSchema($registry);
+
+    // Instantiate the schema and add all extensions.
+    return parent::getSchema($registry);
+  }
+
+  /**
+   * Provides a base schema for Open Social.
+   *
+   * This ensures that other modules have common types available to them to
+   * build on.
+   *
+   * @param \Drupal\graphql\GraphQL\ResolverRegistryInterface $registry
+   *   The resolver registry.
+   */
+  protected function getBaseSchema(ResolverRegistryInterface $registry) {
+    $builder = new ResolverBuilder();
+
+    // TextFormat fields.
+    $registry->addFieldResolver('TextFormat', 'name',
```

Comment:
I think this is no longer valid :trollface:

Reply:
@ronaldtebrake Actually it is! We currently expose the text format for something to the client: https://github.com/goalgorilla/open_social/pull/1994/files#diff-0490ff176a1c7da94e6c07fc1f954fe341e8c0e003e4314ca2006ac2595bd877R167 although we may have to find a better way to provide clients with these details without having to be familiar with the meaning of our text formats.